File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-4.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length == 1) return 1;

        int uniqueIndex = 1;
        int min = nums[0];
        for(int i = 1; i < nums.length; i++){
           if(nums[i] > min){
                nums[uniqueIndex] = nums[i];
                uniqueIndex++;
                min = nums[i];
           }
        }
        for(int i = uniqueIndex +1; i < nums.length; i++){
            nums[i] = 0;
        }

        return uniqueIndex;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        return self.findMinPtr(nums, 0, len(nums) - 1)

    def findMinPtr(self, nums, lptr, rptr) -> int:
        
        if lptr == rptr:
            return nums[lptr]

        mid = (rptr + lptr) // 2
        left = self.findMinPtr(nums, lptr, mid)
        right = self.findMinPtr(nums, mid + 1, rptr)

        return min(left, right)
